server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java

```diff
@@ -246,6 +249,110 @@ public String toString() {
         assertConnectedExactlyToNodes(transportService, targetNodes);
     }
 
+    public void testDisconnectionHistory() {
+        final DeterministicTaskQueue deterministicTaskQueue = new DeterministicTaskQueue();
+        final ThreadPool threadPool = deterministicTaskQueue.getThreadPool();
+        final long reconnectIntervalMillis = CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.get(Settings.EMPTY).millis();
+        final TimeValue reconnectIntervalTimeValue = TimeValue.timeValueMillis(reconnectIntervalMillis);
```

Comment:
nit: suggest doing these in the opposite order:

```suggestion
        final TimeValue reconnectIntervalTimeValue = CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.get(Settings.EMPTY);
        final long reconnectIntervalMillis = reconnectIntervalTimeValue.millis();
```